src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1300,11 +1311,10 @@ private int getMostEfficientTool(@NotNull final Block target)
         for (int i = 0; i < new InvWrapper(worker.getInventoryCitizen()).getSlots(); i++)
         {
             final ItemStack item = inventory.getStackInSlot(i);
-            final int level = Utils.getMiningLevel(item, tool);
-
+            final int level = Utils.getMiningLevel(item, toolType);
             if (level >= required && level < bestLevel)
             {
-                if (tool == null || InventoryUtils.verifyToolLevel(item, level, hutLevel))
+                if (toolType == ToolType.NONE || InventoryUtils.verifyToolLevel(item, level, required, hutLevel))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)